ip-utils/src/main/java/org/hypertrace/config/utils/ip/IpValidationUtils.java

```diff
@@ -0,0 +1,163 @@
+package org.hypertrace.config.utils.ip;
+
+import static java.math.BigInteger.ONE;
+
+import inet.ipaddr.IPAddress;
+import inet.ipaddr.IPAddressString;
+import inet.ipaddr.IPAddressStringParameters;
+import java.util.List;
+import lombok.Getter;
+import org.apache.commons.validator.routines.RegexValidator;
+
+/**
+ * Using 3rd party library <a
+ * href="https://github.com/seancfoley/IPAddress">seancfoley/IPAddress</a>. Validates an ipv6 based
+ * on formats specified over <a
+ * href="https://seancfoley.github.io/IPAddress/IPAddress/apidocs/inet/ipaddr/IPAddressString.html">here</a>.
+ */
+public class IpValidationUtils {
+  private static final String IPV4_REGEX = "^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})$";
+  private static final RegexValidator IPV4_VALIDATOR = new RegexValidator(IPV4_REGEX);
+  private static final int STANDARD_IPV4_LENGTH = 15;
+
+  // modify here to exclude/include certain address formats
+  private static final IPAddressStringParameters ADDRESS_VALIDATION_PARAMS =
+      new IPAddressStringParameters.Builder()
+          // Allows ipv4 joined segments like 1.2.3, 1.2, or just 1 For the case of just 1 segment
+          .allow_inet_aton(false)
+          // Allows an address to be specified as a single value, eg ffffffff, without the standard
+          // use of segments like 1.2.3.4 or 1:2:4:3:5:6:7:8
+          .allowSingleSegment(false)
+          // Allows zero-length IPAddressStrings like ""
+          .allowEmpty(false)
+          .toParams();
+
+  /**
+   * Reason for converting to prefix block explained over <a
+   * href="https://github.com/seancfoley/IPAddress/wiki/Code-Examples-2:-Subnet-Containment,-Matching,-Comparing#containment-within-the-enclosing-cidr-prefix-block">here</a>
+   */
+  public static boolean isValidSubnet(String subnet) {
+    IPAddress parsedAddress = parseAddress(subnet);
+    if (parsedAddress == null) {
+      return false;
+    }
+    return !parsedAddress
```

Comment:
I'm not sure what you're trying to do here, if it parses we should allow it - we don't care if it represents one or multiple. Is this doing something else? If so please include a more descriptive comment rather than a link.